test/functional/unified-spec-runner/operations.ts

```diff
@@ -294,19 +328,36 @@ operations.set('insertMany', async ({ entities, operation }) => {
 });
 
 operations.set('iterateUntilDocumentOrError', async ({ entities, operation }) => {
-  const changeStream = entities.getEntity('stream', operation.object);
-  // Either change or error promise will finish
-  return Promise.race([
-    changeStream.eventCollector.waitAndShiftEvent('change'),
-    changeStream.eventCollector.waitAndShiftEvent('error')
-  ]);
+  try {
+    const changeStream = entities.getEntity('stream', operation.object);
+    // Either change or error promise will finish
+    return Promise.race([
+      changeStream.eventCollector.waitAndShiftEvent('change'),
+      changeStream.eventCollector.waitAndShiftEvent('error')
+    ]);
+  } catch (e) {
+    const findCursor = entities.getEntity('cursor', operation.object);
+    return await findCursor.next();
+  }
+});
+
+operations.set('listCollections', async ({ entities, operation }) => {
+  const db = entities.getEntity('db', operation.object);
+  const { filter, batchSize } = operation.arguments;
```

Comment:
are we intentionally ignoring potential other options? I am concerned that this approach to implementing commands (here and at `listIndexes` in L355) might cause incorrect tests sneak in when we sync specs that introduce new arguments, I feel like a better approach would be to do something like `{ filter, ...opts } = operation.arguments` and then pass them in as `(filter, opts)`, with any future unsupported parameters explicitly omitted (e.g., `{ filter, unsupportedOpt, ...opts } = operation.arguments`)

Reply:
Updated to pass the options through.